robot/web/web.go

```diff
@@ -454,7 +454,10 @@ func (svc *webService) Reconfigure(ctx context.Context, deps resource.Dependenci
 	if err := svc.updateResources(deps); err != nil {
 		return err
 	}
-	return svc.addNewStreams(ctx)
+	if !svc.isRunning || svc.cancelCtx == nil {
```

Comment:
[opt] We should probably rely only on `svc.isRunning` and assume that all other members are initialized. If they aren't, that's a real bug, and the caller of `svc.cancelCtx` will segfault with a stack trace which allows us to find it quickly. As written, the service could be running and, for some odd reason, not set `svc.cancelCtx` and not error or produce any warnings. It'll silently never reconfigure.

Reply:
updated